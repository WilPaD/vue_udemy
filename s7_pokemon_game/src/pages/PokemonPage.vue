<template>
    <h1>¿ Quién es este Pokemon ?</h1>
    <!-- Imagen Pokemon -->
    <PokemonPicture :pokemonId="4" :showPokemon="true" />

    <!-- Opciones Pokemon -->
    <PokemonOptions />
    
</template>

<script>

    import PokemonOptions from '@/components/PokemonOptions.vue';
    import PokemonPicture from '@/components/PokemonPicture.vue';
    
    import getPokemonOptions from '@/helpers/getPokemonOptions'

    export default {
        components:{    
            PokemonPicture,
            PokemonOptions
        }
        
    }

</script>

<style>

</style>
