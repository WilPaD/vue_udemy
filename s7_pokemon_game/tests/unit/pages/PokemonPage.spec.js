import { shallowMount, mount } from "@vue/test-utils";
import PokemonPage from "@/pages/PokemonPage"
import { pokemons } from "../mocks/pokemons.mock";

describe('PokemonPage Component', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallowMount(PokemonPage)
    })

    
    test('Debe hacer match con el snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
    

    test('debe de llamar el mixPokemonArray al montar', () => {
        const mixPokemonArraySpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArray')
        wrapper = shallowMount( PokemonPage )

        expect( mixPokemonArraySpy ).toHaveBeenCalled()
    })

    test('debe hacer match con el snapshot cuando salgan los pokemons', () => {
        //El mount crea el componente y monta todo el componente
        //El shallowMount monta sutilmente el componente sin todo el ciclo de vida
        const wrapper = shallowMount( PokemonPage, {
            data(){
                return{
                    pokemonArr: pokemons,
                    pokemon: pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
    })


    test('Debe de mostrar los componentes PokemonPicture y PokemonOptions', () => {

        /** Deben existir
         * PokemonPicture
         * PokemonOption
         */

        //PokemonPicture attribute pokemonId === 1
        //PokemonOptions attribute pokemons toBe true
        const wrapper = shallowMount( PokemonPage, {
            data(){
                return{
                    pokemonArr: pokemons,
                    pokemon: pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })
        const picture = wrapper.find('pokemon-picture-stub')
        const options = wrapper.find('pokemon-options-stub')

        expect( picture.exists() ).toBeTruthy()
        expect( options.exists() ).toBeTruthy()

        
        expect( picture.attributes('pokemonid') ).toBe('1')
        expect( options.attributes('pokemons') ).toBeTruthy()
    })










})