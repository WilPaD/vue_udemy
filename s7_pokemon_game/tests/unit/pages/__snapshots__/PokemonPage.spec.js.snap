// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component Debe hacer match con el snapshot 1`] = `<h1> Espere por favor </h1>`;

exports[`PokemonPage Component debe hacer match con el snapshot cuando salgan los pokemons 1`] = `
<div>
  <h1>¿ Quién es este Pokemon ?</h1><!-- Imagen Pokemon -->
  <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub><!-- Opciones Pokemon -->
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
  <!--v-if-->
</div>
`;
